File: Python/No_boring_zeros.py
def no_boring_zeros(n):
    if n == 0 :
        return 0
    
    stop = False
    new_list = []

    print(str(n))

    for v in str(n):
        if (v.isdigit()):
            new_list.append(int(v))
        else:
            new_list.append(v)

    current_list = new_list[::-1]

    while not stop:
        if current_list[0] == 0:
            del current_list[0]
        else:
            stop = True

    result = int("".join(map(str, current_list[::-1])))
    return result


print(no_boring_zeros(-1050))

File: JavaScript/NoBoringZeros.js
function noBoringZeros(n) {
  if (n === 0) return 0;

  const arrString = String(n).split("").reverse();
  let stop = false;
  
  while (!stop) {
    if (arrString[0] === '0') {
      arrString.splice(0, 1);
    } else {
      stop = true;
    }
  };

  const result = +arrString.reverse().join("")
  return result;
}

console.assert(noBoringZeros(-1050) === -105, "erro no teste 1!")